docs/migration_guide.md

```diff
@@ -0,0 +1,51 @@
+# Migrating from scikit-build
+
+```{warning}
+scikit-build-core is under active development.
+This guidance will be updated on a best-effort basis, but if you are working at the bleeding edge some of it may be out of date.
+```
+
+## Config changes
+
+- The `build-system.build-backend` key in pyproject.toml must be changed to
+  `scikit_build_core.build`.
+- Replace `scikit-build` with `scikit-build-core` in `build-system.requires`.
+- You may remove `cmake` and `ninja` from `build-system.requires` (IS THIS
+  REQUIRED? CAN THESE BE LEFT IN PLACE IF DESIRED? WHAT IF STRICTER VERSIONS
+  REQUIREMENTS ARE NEEDED THAN WHAT SCIKIT-BUILD-CORE'S CONFIG SUPPORTS?)
+- You must fill out the `tool.scikit-build` table in pyproject.toml, see
+  [getting started](./getting_started.md) for more information.
+
+## CMake changes
+
+scikit-build users wishing to switch to scikit-build-core should be aware of the
+following changes that must be made to their CMake files:
```

Comment:
One other change is the install directory. scikit-build tries (very buggily, based on issues!) to put things in the package folder, while scikit-build-core doesn't muddle with it and installs to the site-packages folder directly instead. Adding `wheel.install-dir = "<package_name>"` will avoid CMakeLists changes, though it might be better to install to `<package_dir>`.

Reply:
For sure, moving away from that model is definitely an improvement in my book.